### DISCUSSION TITLE
dwarfsextract option (command) of the read operation (process) for yad (progressbar) ?

### DISCUSSION THREAD

**Discussion by fwn0** [mhx/dwarfs#116] [Q&A] [answered]
I would like to have the progress of extracting dwarf archives displayed graphically. Is there any way to send the unpacking process to a pipe?
I have tried something similar 7 zip or fa:
```
# Unpacking package
"$7zz" x archive.7z -aoa -oc:"$1" >> "$progress_pipe" \
2>&1 >> "$progress_pipe" & echo "$!" >> "$main_proc_id"

# Wait here and it returns exit status "${?}"
# It's stderr redirected to /dev/null for prevent messages from the kill
wait "$(<"$main_proc_id")" 2>/dev/null

if [ "$?" = "0" ]; then
	echo "#Installation completed" >> "$progress_pipe"
	kill "$main_pid" >"$main_proc_id" 2>/dev/null
else
	echo "#Installation error" >> "$progress_pipe"
	echo $? >> "$progress_pipe" 
	kill "$main_pid" >"$main_proc_id" 2>/dev/null
fi

or: 
# Unpacking from the archive
"$fa" x "$dir"/archive.fa -o+ -dp"$1" >> "$progress_pipe" \
2>&1 >> "$progress_pipe" & echo "$!" >> "$main_proc_id"

# Wait here and it returns exit status "${?}"
# It's stderr redirected to /dev/null for prevent messages from the kill
wait "$(<"$main_proc_id")" 2>/dev/null

if [ "$?" = "0" ]; then
	echo "#Installation completed" >> "$progress_pipe"
	kill "$main_pid" 2>/dev/null >"$main_proc_id"
else
	echo "#Installation error" >> "$progress_pipe"
	kill "$main_pid" 2>/dev/null >"$main_proc_id"
fi

```
which works fine, but for dwarfsextract it cannot capture this process.

**Comment by mhx** [comment, marked as answer]
Yep, `dwarfsextract` currently has no way of displaying or sending progress information.

It was easy enough to add, though, so it's now in the latest release candidate [dwarfs-0.7.0-RC3](https://github.com/mhx/dwarfs/releases/tag/v0.7.0-RC3).

You need to supply the `--stdout-progress` flag to enable progress output. I've tested this with `zenity`, so I would assume it works just fine with `yad` as well:

    dwarfsextract -i rasp-norm2.dwarfs -o tmp --stdout-progress \
        | zenity --progress --auto-close --no-cancel --time-remaining
